src/Misc/MSet.jl

```diff
@@ -119,48 +568,92 @@ function Base.filter(pred, s::MSet)
   return t
 end
 
+function Base.filter!(pred, s::MSet)
+  un = unique(s)
+  for x in un
+    if !pred(x)
+      delete!(s, x)
+    end
+  end
+  return s
+end
 
 @doc raw"""
-    multiplicities(s::MSet)
+    multiplicities(s::MSet{T}) -> ValueIterator{Dict{T, Int}}
 
-Return an iterator for the multiplicities of all the elements.
+Return an iterator for the multiplicities of all the elements in `s`.
+
+# Examples
+```jldoctest
+julia> m = multiset([1,1,2,3,4,4,5])
+MSet{Int64} with 7 elements:
+  5
+  4 : 2
+  2
+  3
+  1 : 2
+
+julia> mult_m = multiplicities(m)
+ValueIterator for a Dict{Int64, Int64} with 5 entries. Values:
+  1
+  2
+  1
+  1
+  2
+
+julia> collect(mult_m)
+5-element Vector{Int64}:
+ 1
+ 2
+ 1
+ 1
+ 2
+```
 """
 function multiplicities(s::MSet)
   return values(s.dict)
 end
 
 @doc raw"""
-    multiplicity(s::MSet, x)
+    multiplicity(s::MSet{T}, x::T) -> Int
 
-The multiplicity of the element `x`` in the multi-set `s - or zero if
-  `x` is not in `s`,
+Return the multiplicity of the element `x` in the multi-set `s`. If `x` is not
+in `s`, return 0.
+
+# Examples
+```jldoctest
+julia> m = multiset([1,1,2,3,4,4,5])
+MSet{Int64} with 7 elements:
+  5
+  4 : 2
+  2
+  3
+  1 : 2
+
+julia> multiplicity(m, 2)
+1
+
+julia> multiplicity(m, 6)
+0
+```
 """
-function multiplicity(s::MSet{T}, x::T) where {T}
-  y = x isa T ? x : T(x)
-  if haskey(s.dict, y)
-    return s.dict[y]
+function multiplicity(s::MSet, x)
```

Comment:
Change from constant time to linear? Why?

Reply:
Same as explain for the `in` function: if I have a `MSet{QQFieldElem}`, then for testing the multiplicity of 2, one needs to check for `QQ(2)`. If we want to force this and then do not leave room for this kind of flexibility, I am ok with reverting this.